lib/services/upstart.c

```diff
@@ -391,51 +391,67 @@ upstart_mask_error(svc_action_t *op, const char *error)
     return FALSE;
 }
 
+/*!
+ * \internal
+ * \brief Process the completion of an asynchronous job start, stop, or restart
+ *
+ * \param[in] pending    If not NULL, DBus call associated with request
+ * \param[in] user_data  Action that was executed
+ */
 static void
-upstart_async_dispatch(DBusPendingCall *pending, void *user_data)
+job_method_complete(DBusPendingCall *pending, void *user_data)
 {
     DBusError error;
     DBusMessage *reply = NULL;
     svc_action_t *op = user_data;
 
-    dbus_error_init(&error);
-    if(pending) {
+    // Grab the reply
+    if (pending != NULL) {
         reply = dbus_pending_call_steal_reply(pending);
     }
 
+    // Determine result
+    dbus_error_init(&error);
     if (pcmk_dbus_find_error(pending, reply, &error)) {
 
         /* ignore "already started" or "not running" errors */
+        op->rc = PCMK_OCF_UNKNOWN_ERROR;
+        op->status = PCMK_EXEC_ERROR;
         if (!upstart_mask_error(op, error.name)) {
             crm_err("%s for %s: %s", op->action, op->rsc, error.message);
         }
         dbus_error_free(&error);
 
-    } else if (!g_strcmp0(op->action, "stop")) {
-        /* No return vaue */
+    } else if (pcmk__str_eq(op->action, "stop", pcmk__str_none)) {
+        // Call has no return value
         op->rc = PCMK_OCF_OK;
+        op->status = PCMK_EXEC_DONE;
 
```

Comment:
this is successful as well - probably should have a similar debug log as below

Reply:
makes sense